client/me/purchases/remove-purchase/index.jsx

```diff
@@ -117,32 +228,100 @@ const RemovePurchase = React.createClass( {
 			productName = getName( purchase );
 
-		if ( isDomainRegistration( purchase ) ) {
-			return (
-				<p>
-					{
-						this.translate( 'This will remove %(domain)s from your account.', {
-							args: { domain: productName }
-						} )
-					}
-					{ ' ' }
-					{ this.translate( 'By removing, you are canceling the domain registration. This may stop you from using it again, even with another service.' ) }
-				</p>
-			);
+		return (
+			<p>
+				{
+					this.translate( 'This will remove %(domain)s from your account.', {
+						args: { domain: productName }
+					} )
+				}
+				{ ' ' }
+				{ this.translate( 'By removing, you are canceling the domain registration. This may stop you from using it again, even with another service.' ) }
+			</p>
+		);
+	},
+
+	renderPlanDialogs() {
+		const buttons1 = [
+			{
+				action: 'cancel',
+				disabled: this.state.isRemoving,
+				label: this.translate( 'Cancel' )
+			},
+			{
+				action: 'next-2',
+				disabled: this.state.isRemoving || this.state.questionOneRadio == null || this.state.questionTwoRadio == null,
+				label: this.translate( 'Next' ),
+				onClick: this.displaySurveyTwo
+			} ],
+			buttons2 = [
+				{
+					action: 'cancel',
+					disabled: this.state.isRemoving,
+					label: this.translate( 'Cancel' )
+				},
+				{
+					action: 'prev-1',
+					disabled: this.state.isRemoving,
+					label: this.translate( 'Previous' ),
+					onClick: this.displaySurveyOne
+				},
+				{
+					action: 'remove',
+					disabled: this.state.isRemoving,
+					isPrimary: true,
+					label: this.translate( 'Remove' ),
+					onClick: this.removePurchase
+				} ],
+			productName = getName( getPurchase( this.props ) );
+
+		if ( ! config.isEnabled( 'upgrades/removal-survey' ) ) {
+			buttons2.splice( 1, 1 );
 		}
 
-		let includedDomainText;
+		return (
+			<div>
+				<Dialog
+					buttons={ buttons1 }
+					className="remove-purchase__dialog"
+					isVisible={ this.state.isDialogVisible && this.state.isSurveyOneVisible }
+					onClose={ this.closeDialog }
+					transitionEnter={ false }
+					transitionLeave={ false }>
+					<FormSectionHeading>{ this.translate( 'Remove %(productName)s', { args: { productName } } ) }</FormSectionHeading>
+					{ this.renderQuestionOne() }
+					{ this.renderQuestionTwo() }
+				</Dialog>
+				<Dialog
+					buttons={ buttons2 }
+					className="remove-purchase__dialog"
+					isVisible={ this.state.isDialogVisible && this.state.isSurveyTwoVisible }
+					onClose={ this.closeDialog }
+					transitionEnter={ false }
+					transitionLeave={ false }>
+					<FormSectionHeading>{ this.translate( 'Remove %(productName)s', { args: { productName } } ) }</FormSectionHeading>
+					{ config.isEnabled( 'upgrades/removal-survey' ) ? this.renderFreeformQuestion() : null }
+					{ this.renderPlanDialogsText() }
+				</Dialog>
+			</div>
+		);
+	},
 
+	renderPlanDialogsText() {
+		const purchase = getPurchase( this.props ),
+			productName = getName( purchase );
+
+		let includedDomainText;
```

Comment:
Why are we setting this to a variable? It never changes, we just conditionally include it. I think it would be a little cleaner if we just used the original conditional to determine whether to include the text and avoid a variable completely. Like so:
```js
<p>always shown</p>
{ ( someConditional ) && <p>conditionally shown</p> }
```